server/routes/bookVisit/mainContactController.ts

```diff
@@ -9,41 +10,39 @@ export default class MainContactController {
     return async (req, res) => {
       const { selectedVisitors } = req.session.bookingJourney
 
-      const adults = selectedVisitors.reduce((adultVisitors: Visitor[], visitor: Visitor) => {
-        if (visitor.adult ?? true) {
-          adultVisitors.push(visitor)
-        }
-
-        return adultVisitors
-      }, [])
+      const adultVisitors = selectedVisitors.filter(visitor => visitor.adult)
 
       res.render('pages/bookVisit/mainContact', {
         errors: req.flash('errors'),
         formValues: req.flash('formValues')?.[0] || {},
-        adultVisitors: adults,
+        adultVisitors,
       })
     }
   }
 
   public submit(): RequestHandler {
     return async (req, res) => {
-      const { bookingJourney } = req.session
       const errors = validationResult(req)
-
       if (!errors.isEmpty()) {
         req.flash('errors', errors.array())
         req.flash('formValues', req.body)
         return res.redirect(`/book-visit/main-contact`)
       }
 
-      const selectedContact = bookingJourney.selectedVisitors.find(
-        (visitor: Visitor) => req.body.contact === visitor.visitorDisplayId.toString(),
-      )
+      const { bookingJourney } = req.session
+      const { contact }: { contact: string } = req.body
+
+      if (contact === 'someoneElse') {
+        bookingJourney.mainContact = { contact: req.body.someoneElseName }
+      } else {
+        const contactVisitor = bookingJourney.selectedVisitors.find(
+          visitor => visitor.visitorDisplayId.toString() === contact,
+        )
+        bookingJourney.mainContact = { contact: contactVisitor }
+      }
 
-      bookingJourney.mainContact = {
-        contact: selectedContact,
-        phoneNumber: req.body.hasPhoneNumber === 'yes' ? req.body.phoneNumber : undefined,
-        contactName: selectedContact === undefined ? req.body.someoneElseName : undefined,
+      if (req.body.hasPhoneNumber === 'yes') {
+        bookingJourney.mainContact.phoneNumber = req.body.phoneNumber
```

Comment:
without having the conditional to set to undefined, when revisiting this page - could you not end up with hasPhoneNumber = no, but booking journey having a phone number, so bad data?

Reply:
Hmmm 🤔 - good question. Just tested out and I think it is OK because on line 34 (or 39), whatever is in `bookingJourney.mainContact` is replaced with a new object so any previous data is gone.

We've got other tickets to look at pre-populating each page in the journey when you go back via 'Check visit' page so these kinds of scenarios will be good to test as part of that.

Reply:
good point - replacing the whole object on post should cleanly work around this possibility, so a non-issue in this case

When we re-visit these pages on the pre-populate tickets these are the thing's we have experience testing for anyway so should be good to go